byterun/dynlink.c

```diff
@@ -130,14 +130,13 @@ static void open_shared_lib(char_os * name)
   void * handle;
 
   realname = caml_search_dll_in_path(&caml_shared_libs_path, name);
-  caml_gc_message(0x100, "Loading shared library %"
-                  ARCH_CHARNATSTR_PRINTF_FORMAT "\n", realname);
+  caml_gc_message(0x100, "Loading shared library %s\n", caml_utf8_string_of_os(realname));
```

Comment:
I agree with @gasche's concern about using a static buffer. You cover nearly all cases of the strdup with either caml_fatal_arg (which will automatically clean up caml_stat_strdup anyway) or caml_gc_message. You could introduce caml_gc_message_free which strictly takes a string parameter and frees it after displaying the message.